File: sdk/src/ReserveFairValue.ts
import { SlotTracking } from "./SlotTracking";

export class ReserveFairValue {
    value: number;
    slotTracking: SlotTracking;

    constructor(value: number, slotTracking: SlotTracking) {
        this.value = value;
        this.slotTracking = slotTracking
    }
}

File: programs/vyper-core/src/state/reserve_fair_value.rs
use anchor_lang::prelude::*;

use super::SlotTracking;

#[repr(C, align(8))]
#[derive(AnchorDeserialize, AnchorSerialize, Clone, Copy, Debug, Default)]
pub struct ReserveFairValue {
    /// reserve fair value expressed in bps
    pub value: [[u8; 16]; 10],
    pub slot_tracking: SlotTracking,
}

impl ReserveFairValue {
    pub const LEN: usize = 16*10 + // pub value: [[u8; 16]; 10],
    SlotTracking::LEN; // pub slot_tracking: SlotTracking
}
